packages/dev/core/src/Materials/Textures/videoTexture.ts

```diff
@@ -267,15 +268,13 @@ export class VideoTexture extends Texture {
 
         return video;
     }
+    private _videoResizeEventListener = (): void => {
+        this._resizeInternalTexture(false /* skipDispose */);
+    }
 
-    private _createInternalTexture = (): void => {
-        if (this._texture != null) {
-            if (this._displayingPosterTexture) {
-                this._texture.dispose();
-                this._displayingPosterTexture = false;
-            } else {
-                return;
-            }
+    private _resizeInternalTexture = (skipDispose: boolean): void => {

```

Comment:
Does it work if we don't have this flag and instead dispose outside of this function when needed?

Reply:
Maybe I'm missing something, but why is this param needed at all? Couldn't this function just always dispose `_texture` if it is not null, and then just set `_texture` to null right after it is disposed in `_createInternalTexture` (line 297), or is there some other logic that depends on the presence of a disposed texture?

Reply:
I went with @ryantrem proposal (mostly) as it also addresses @sebavan comment where `_resizeInternalTexture` could cause a leak.

Now `_resizeInternalTexture` does not take any arguments and will always dispose the texture (if it exists) before setting the new one.

`_createInternalTexture` was updated to no longer call `_texture.dispose()` and instead let `_resizeInternalTexture` do that. The conditional logic already worked that way in `_createInternalTexture` so the flow of that function remains the same, dispose is just called a few lines later now.